### PR TITLE
DIG-1381: Switch katsu from uwsgi to Gunicorn

### PR DESCRIPTION
**Pull request by SonQBChau**
## What's new
- Use gunicorn for web server
- Remove uwsgi

### How to test
- Only need to test in docker stack
- Rebuild this PR and run test-integration

### CODE REVIEW

File: gunicorn_conf.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import logging.config
+import os
+
+bind = "0.0.0.0"
+workers = 2
+threads = 4
```
Comment (mshadbolt) [inline gunicorn_conf.py]: How did you decide these numbers in the configuration?

Comment (SonQBChau) [reply gunicorn_conf.py]: trials and errors

**Comment by SonQBChau** [comment]
workers and threads are now controllable from ENV

**Comment by mshadbolt** [comment]
This warning appears in the katsu logs when starting:
`2024-05-29 17:10:51 [2024-05-30 00:10:51 +0000] [17] [INFO] ASGI 'lifespan' protocol appears unsupported.`

Is it something we should be concerned about?

**Comment by SonQBChau** [comment]
There are some small bugs I haven't ironed out, like this one. It doesn't seem to cause any issues though

**Review by mshadbolt — APPROVED** [review]
Looks good to me and works as expected with the stack, I cannot trigger the federation SIGPIPE errors that I could before, hopefully this fixes it!

**Comment by SonQBChau** [comment]
Interesting. I didn't test this through federation. Limiting the low number of workers might have a (good) side effect, but we'll need to do more testing to confirm.

**Pull request merged**